src/main/flight/autopilot.c

```diff
@@ -91,14 +158,223 @@ void altitudeControl(float targetAltitudeCm, float taskIntervalS, float vertical
+            // value at this point is a 'best guess' to detect IMU failure in the event the user has no Mag
+            // if entering poshold from a stable hover, we would only exceed this if IMU was disoriented
+            // if entering poshold at speed, it may overshoot this value and falsely fail, if so need something more complex
+        }
+
+        // ** PIDs **
+        // separate PID controllers for latitude (NorthSouth or ns) and longitude (EastWest or ew)
+
+        // ** P **
+
+        const float nsP = nsDistance * positionPidCoeffs.Kp;
+        const float ewP = ewDistance * positionPidCoeffs.Kp;
+
+        // ** I **
+
+        const float leak = 1.0f - 0.4f * posHold.gpsDataIntervalS; // gpsDataIntervalS is not more than 1.0s
+        if (!posHold.isStartingNS){
+            // only accumulate iTerm after completing the start phase
+            // perhaps need a timeout on the start phase ?
+            posHold.integralNS += nsDistance * posHold.gpsDataIntervalS;
+        } else {
+            // while moving sticks, slowly leak iTerm away, approx 2s time constant
+            posHold.integralNS *= leak;
+        }
+        if (!posHold.isStartingEW){
+            posHold.integralEW += ewDistance * posHold.gpsDataIntervalS;
+        } else {
+            posHold.integralEW *= leak;
+        }
+
+        const float nsI = posHold.integralNS * positionPidCoeffs.Ki;
+        const float ewI = posHold.integralEW * positionPidCoeffs.Ki;
+
+        // ** D ** //
+        // get change in distance in NS and EW directions from gps.c using the `GPS_distances` function
+        // this gives cleaner velocity data than the module supplied GPS Speed and Heading information
+
+        const float velocityNS = (nsDistance - posHold.previousDistanceNS) * posHold.gpsDataFreqHz; // cm/s, minimum step 11.1 cm/s
+        const float velocityEW = (ewDistance - posHold.previousDistanceEW) * posHold.gpsDataFreqHz;
+        posHold.previousDistanceNS = nsDistance;
+        posHold.previousDistanceEW = ewDistance;
+
+        float accelerationNS = (velocityNS - posHold.previousVelocityNS) * posHold.gpsDataFreqHz;
+        posHold.previousVelocityNS = velocityNS;
+        float accelerationEW = (velocityEW - posHold.previousVelocityEW) * posHold.gpsDataFreqHz;
+        posHold.previousVelocityEW = velocityEW;
+
+        // scale and filter - filter cutoffs vary during the startup phase
+        float nsD = velocityNS * positionPidCoeffs.Kd;
+        pt1FilterUpdateCutoff(&velocityNSLpf, posHold.pt1Gain);
+        nsD = pt1FilterApply(&velocityNSLpf, nsD);
+
+        float ewD = velocityEW * positionPidCoeffs.Kd;
+        pt1FilterUpdateCutoff(&velocityEWLpf, posHold.pt1Gain);
+        ewD = pt1FilterApply(&velocityEWLpf, ewD);
+
+        float nsA = accelerationNS * positionPidCoeffs.Kf;
+        pt2FilterUpdateCutoff(&accelerationNSLpf, posHold.pt1Gain); // using PT1 gain for stronger cutoff
```

Comment:
Use lower cutoff instead

Also, you can apply velocity filter to velocity and calculate acceleration from filtered value. That way you can avoid repeating (identical) fitler here

Reply:
Originally I did exactly that.  I applied one PT1 to velocity, took acceleration from the filtered velocity, and applied a second PT1.  
At this point in the code development, these values came from the distance to home vector length, so I was concerned about delay.
Now the model has changed, concerns about delay from some earlier vector angle are not relevant, so I probably can go back to the simpler code with no loss of performance.  I anticipate doing that.
Thanks for reminding me.